scripts/generate_pr_description.py

```diff
@@ -0,0 +1,108 @@
+import json
+import os
+import requests
+import openai
+
+MAX_CHARS = 400000  # Maximum characters for changes summary
+
+def truncate_changes(changes_summary):
+    """Truncates the changes summary to fit within MAX_CHARS."""
+    total_chars = 0
+    truncated_summary = []
+    for change in changes_summary:
+        change_chars = len(change)
+        if total_chars + change_chars > MAX_CHARS:
+            remaining_chars = MAX_CHARS - total_chars
+            if remaining_chars > 50:  # Ensure we're not adding just a few characters
+                truncated_change = change[:remaining_chars]
+                truncated_summary.append(truncated_change + "...")
+            break
+        total_chars += change_chars
+        truncated_summary.append(change)
+    return truncated_summary
+
+def generate_pr_description():
+    # GitHub API setup
+    token = os.environ['GITHUB_TOKEN']
+    repo = os.environ['GITHUB_REPOSITORY']
+    
+    # Get PR number from the event payload
+    with open(os.environ['GITHUB_EVENT_PATH']) as f:
+        event = json.load(f)
+    pr_number = event['pull_request']['number']
+    
+    headers = {'Authorization': f'token {token}'}
+    api_url = f'https://api.github.com/repos/{repo}/pulls/{pr_number}'
+
+    # Get current PR description
+    pr_info = requests.get(api_url, headers=headers).json()
+    current_description = pr_info['body'] or ''
+
+    # Check if description matches the default template
+    default_template_indicator = "I implemented/fixed _ to achieve _."
+
+    if default_template_indicator in current_description:
+        # Get PR files
+        files_url = f'{api_url}/files'
+        files = requests.get(files_url, headers=headers).json()
+
+        # Process files
+        changes_summary = []
+        for file in files:
+            filename = file['filename']
+            status = file['status']
+
+            if status == 'added':
```

Comment:
Usually, if there is a new file, there is some meaningful code in there. That's why it might be nice to add the `patch` for the new files as well.